File: backend/internal/model/customer.go
package model

import "time"

// Customer 对应 customer 表。
type Customer struct {
	ID           int64
	OwnerUserID  int64
	CustomerName string
	Phone        string
	Address      string
	BusinessType string
	Status       int8
	Remark       string
	CreateTime   time.Time
	UpdateTime   time.Time
}

// ContractTemplate 对应 contract_template 表。
type ContractTemplate struct {
	ID               int64
	OwnerUserID      int64
	TemplateName     string
	OriginalFileName string
	Status           int8
	CurrentVersionID *int64
	CurrentVersionNo int
	Description      string
	CreateTime       time.Time
	UpdateTime       time.Time
}

// ContractTemplateVersion 对应 contract_template_version 表。
type ContractTemplateVersion struct {
	ID              int64
	TemplateID      int64
	VersionNo       int
	CreatedBy       int64
	OssObjectKey    string
	OssURL          string
	FileName        string
	FileSize        int64
	FileHash        string
	DocumentContent string
	ChangeSummary   string
	CreateTime      time.Time
}

// ContractTemplateDetail 模板详情，含当前版本。
type ContractTemplateDetail struct {
	Template ContractTemplate
	Version  *ContractTemplateVersion
}

File: frontend/src/types/customer.ts
/** 客户相关类型 */

export interface Customer {
  id: number
  customer_name: string
  phone: string
  address: string
  business_type: string
  remark: string
  create_time: string
  update_time: string
}

export interface CustomerListParams {
  page?: number
  page_size?: number
  keyword?: string
}

export interface CustomerListResult {
  list: Customer[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface CreateCustomerParams {
  customer_name: string
  phone: string
  address?: string
  business_type?: string
  remark?: string
}

export interface CreateContractFromTemplateParams {
  template_id: number
  contract_name?: string
}
